File: ThreatModelDfdService/Model/Entity/DataFlow.cs
namespace ThreatModelDfdService.Model.Entity;

public class DataFlow : BaseEntity
{
    public string Name { get; set; }
    public string Description { get; set; }
    
    public long? SourceElementId { get; set; }
    public virtual DfdElement SourceElement { get; set; }

    public long? TargetElementId { get; set; }
    public virtual DfdElement TargetElement { get; set; }

    public decimal SourcePosition { get; set; }
    public decimal TargetPosition { get; set; }


    public long DfdId { get; set; }
    public virtual Dfd Dfd { get; set; }
}

File: dfd-service/src/main/java/com/tcc/dfd_service/entity/DataFlow.java
package com.tcc.dfd_service.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.math.BigDecimal;

@Entity
@Table(name = "data_flows")
@Getter
@Setter
@NoArgsConstructor
public class DataFlow extends BaseEntity {

    @Column(nullable = false)
    private String name;

    @Column(columnDefinition = "TEXT")
    private String description;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "source_element_id")
    private DfdElement sourceElement;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "target_element_id")
    private DfdElement targetElement;

    @Column(nullable = false)
    private BigDecimal sourcePosition;

    @Column(nullable = false)
    private BigDecimal targetPosition;

    @Column(name = "dfd_id")
    private Long dfdId;
}